discord/experiment.py

```diff
@@ -0,0 +1,217 @@
+"""
+The MIT License (MIT)
+
+Copyright (c) 2015-present Rapptz
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+
+from enum import Enum
+from typing import TYPE_CHECKING, List, Optional, Tuple
+
+if TYPE_CHECKING:
+    from .types.experiment import (
+        GuildExperiment as RawExperiment,
+        Override,
+        Population,
+        UserExperimentAssignment as RawAssignment,
+    )
+
+
+try:
+    from mmh3 import hash as _hash # type: ignore -- this is caught anyway so its fine
+
+    def hash(data: str): # type: ignore -- this function is defined only if it cannot be imported
+        return _hash(data, signed=False)
+except ImportError:
+    from .utils import hash as _hash
+
+    def hash(data: str):
+        return _hash(data)
+
+class FilterTypes(Enum):
```

Comment:
This should be called `ExperimentFilterType`, have lowercase names, and be in enums.py